subsys/rtio/rtio_executor_concurrent.c

```diff
@@ -223,17 +223,29 @@ void rtio_concurrent_ok(struct rtio_iodev_sqe *iodev_sqe, int result)
 		exc->task_status[task_id] |= CONEX_TASK_COMPLETE;
 	}
 
-	bool transaction = sqe->flags & RTIO_SQE_TRANSACTION;
+	bool transaction;
+
+	do {
```

Comment:
This is neat, so the transaction in this case produces completions on behalf of the iodev rather than the iodev having to produce them itself?

Reply:
Correct, I wanted to avoid having any client side changes.